### PR TITLE
HPCC-25493 Always create threads at default priority, not inherited from parent

### PR DESCRIPTION
**Pull request by richardkchapman**
Much less error-prone.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by mckellyln — APPROVED** [review]
This looks fine.  One minor note about not needing the getschedparam call.

File: system/jlib/jthread.cpp [hunk 1]
```diff
@@ -401,6 +401,12 @@ void Thread::startRelease()
                 pthread_attr_setstacksize(&attr, LINUX_STACKSIZE_CAP);
 #endif
         }
+        sched_param param;
+        pthread_attr_getschedparam(&attr, &param);
```
[the remark this answers is not in the document]
Comment (richardkchapman) [reply system/jlib/jthread.cpp:405]: Is that always true for all systems (now and in the future). The get-then-modify pattern seemed to be the recommended way from my quick searching...

File: system/jlib/jthread.cpp [hunk 2]
```diff
@@ -401,6 +401,12 @@ void Thread::startRelease()
...
+        param.sched_priority = 0;
+        pthread_attr_setschedparam(&attr, &param);
+        pthread_attr_setschedpolicy(&attr, SCHED_OTHER);
```
Comment (jakesmith) [inline system/jlib/jthread.cpp:408]: If the process has been started with a different riority, and therefore main is at that priority,
does this set new threads to the same priority as the process/main thread, or to a system standard (SCHED_NORMAL) priority?

**Review by jakesmith — COMMENTED** [review]
@richardkchapman - 1 question.

File: system/jlib/jthread.cpp (hunk 2 above)
Comment (richardkchapman) [reply system/jlib/jthread.cpp:408]: Testing suggests it ends up same priority as the main thread.

**Comment by richardkchapman** [comment]
@jakesmith Tested via a standalone snippet on a raspberry Pi, it seems to work as desired.

**Review by jakesmith — APPROVED** [review]
@richardkchapman looks good to go.

**Comment by richardkchapman** [comment]
@ghalliday Please review/merge.

It's possible to avoid this bug in versions < 8.0 by changing a Roxie setting (to a value that is recommended for production systems anyway) so I think no need to backport

**Pull request merged**